runtime/commonMain/src/kotlinx/serialization/PlatformUtils.kt

```diff
@@ -4,13 +4,14 @@
 
 package kotlinx.serialization
 
-import kotlinx.serialization.internal.defaultSerializer
+import kotlinx.serialization.internal.primitiveSerializerOrNull
 import kotlin.reflect.KClass
 
 @ImplicitReflectionSerializer
-fun <T : Any> KClass<T>.serializer(): KSerializer<T> = compiledSerializer() ?: defaultSerializer()
+fun <T : Any> KClass<T>.serializer(): KSerializer<T> = compiledSerializer() ?: primitiveSerializerOrNull()
     ?: throw SerializationException("Can't locate argument-less serializer for $this. For generic classes, such as lists, please provide serializer explicitly.")
 
+// TODO internals where it is possible
 @ImplicitReflectionSerializer
 expect fun <T : Any> KClass<T>.compiledSerializer(): KSerializer<T>?
```

Comment:
Not yet, will fix in a separate PR.
TODOs are just to have a fast search mechanism

Reply:
I'm wondering, is `primitiveSerializerOrNull` the right name. Perhaps it could be `builtinSerializerOrNull` so that it can support some other well-known types (at least `StringSerializer`)

Reply:
good point, `builtinSerializerOrNull` better reflects both semantics and implementation